src/Ucommerce.Avenue.Umbraco/Controllers/CategoryController.cs

```diff
@@ -63,10 +63,11 @@ private IList<ProductViewModel> MapProductsInCategories(Category category)
             IList<Facet> facetsForQuerying = System.Web.HttpContext.Current.Request.QueryString.ToFacets();
             var productsInCategory = new List<ProductViewModel>();
 
-            foreach (var subCategoryGuid in category.Categories)
+            var subCategories = CatalogLibrary.GetCategories(category.Categories);
+
+            foreach (var subCategory in subCategories)
             {
-                var subCategory = CatalogLibrary.GetCategory(subCategoryGuid);
-                productsInCategory.AddRange(MapProductsInCategories(subCategory));
+                productsInCategory.AddRange(MapProducts(SearchLibrary.GetProductsFor(subCategory.Guid, facetsForQuerying).Results));
```

Comment:
This is still, technically, a SELECT N+1.
All retrieval should be outside the `foreach`.

Reply:
Let's get it in, and I'll help fix this later.